### PR TITLE
Migrate to hardhat and add forwarder fee in contract

### PR DESCRIPTION
**Pull request by sembrestels**

### CODE REVIEW

File: packages/an-delay-app/contracts/Delay.sol [hunk 1]
```diff
@@ -46,7 +47,7 @@ contract Delay is AragonApp, IForwarder, IForwarderFee {
     uint256 public feeAmount;
     address public feeDestination;
 
-    event DelayedScriptStored(uint256 scriptId);
+    event DelayedScriptStored(uint256 scriptId, bytes evmCallScript);
```
Comment (PJColombo) [inline packages/an-delay-app/contracts/Delay.sol]: Could we add the `feeAmount` and `executionTime` as params too?

The only way to get these data is by calling the contract which could be in a state different than the one the contract was in when the delayed script was created

**Review by PJColombo — CHANGES_REQUESTED** [review]
Also, is it necessary to include the previous value when emitting the `ChangeFeeAmount` and `ChangeFeeDestination` events?

We could get them from the previous event

**Review by PJColombo — APPROVED** [review]
LGTM!

**Review by 0xGabi — COMMENTED** [review]
What do you think of using the https://github.com/1Hive/hardhat-aragon plugging to start using it to publish apps. Here is a PR that shows how to do it: https://github.com/1Hive/token-manager-app/pull/20

**Pull request merged**